controller/composite/metacontroller.go

```diff
@@ -168,10 +168,21 @@ func (mc *Metacontroller) sync(key string) error {
 		}
 		return nil
 	}
+
 	if err != nil {
 		mc.eventRecorder.Eventf(cc, v1.EventTypeNormal, events.ReasonSyncError, "[%s] Sync error - %s", cc.Name, err)
 		return err
 	}
+	parentClient, err := mc.dynClient.Resource(cc.Spec.ParentResource.APIVersion, cc.Spec.ParentResource.Resource)
+	if err != nil {
+		return err
+	}
+	if hasStatusSubresource := parentClient.APIResource.HasSubresource("status"); !hasStatusSubresource {
+		mc.eventRecorder.Eventf(cc, v1.EventTypeWarning, events.ReasonSyncError, "[%s] Sync error - ignoring, parent resource %s does not have subresource 'Status' enabled", cc.Name, parentClient.GroupVersionKind())
+		klog.InfoS("Ignoring CompositeController", "name", name, "reason", fmt.Sprintf("subresource 'Status' not enabled for %s", parentClient.GroupVersionKind()))
+		// returning, as we cannot do anything until this 'Status' subresource is added to parent resource
+		return nil
```

Comment:
Can we limit the number of chars in a line to be within 80 chars.
It is unreadable.
In addition, we can use some of the go naming conventions which is quite idiomatic to reduce the line length.
e.g.
```go
if found := parentClient.APIResource.HasSubResource("status"); !found {
  mc.eventRecorder.Eventf(
    cc,
    v1.EventTypeWarning,
    events.ReasonSyncError,
    "[%s] sync error - ignoring ...",
    cc.Name, 
    parentClient.GroupVersionKind(),
  )
  klog.InfoS(
    "Ignoring CompositeController", 
    "name", name, 
    "reason", fmt.Sprintf("subresource 'Status' not enabled for %s", parentClient.GroupVersionKind()),
  )
  return nil
}
```

Reply:
sure, will do it